intg-denonavr/denon_remote.py

```diff
@@ -0,0 +1,820 @@
+            features,
+            attributes={
+                Attributes.STATE: receiver.state,
+            },
+            simple_commands=self._denon_media_player.get_supported_commands(),
+            button_mapping=REMOTE_BUTTONS_MAPPING,
+            ui_pages=DenonRemote._get_remote_ui_pages(),
+        )
+
+    async def command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        """
+        Remote entity command handler.
+
+        Called by the integration-API if a command is sent to a configured remote entity.
+
+        :param cmd_id: command
+        :param params: optional command parameters
+        :return: status code of the command request
+        """
+        if params:
+            repeat = self._get_int_param("repeat", params, 1)
+        else:
+            repeat = 1
+
+        is_succsess = False
+        for _ in range(0, repeat):
+            is_succsess |= await self._handle_command(cmd_id, params) == StatusCodes.OK
+        return StatusCodes.OK if is_succsess else StatusCodes.BAD_REQUEST
+
+    @staticmethod
+    def state_from_avr(avr_state: avr.States) -> ucapi.remote.States:
+        """
+        Convert AVR state to UC API remote state.
+
+        :param avr_state: Denon/Marantz AVR state
+        :return: UC API remote state
+        """
+        if avr_state in REMOTE_STATE_MAPPING:
+            return REMOTE_STATE_MAPPING[avr_state]
+        return ucapi.remote.States.UNKNOWN
+
+    def filter_changed_attributes(self, update: dict[str, Any]) -> dict[str, Any]:
+        """
+        Filter the given attributes and return only the changed values.
+
+        :param update: dictionary with attributes.
+        :return: filtered entity attributes containing changed attributes only.
+        """
+        attributes = {}
+
+        if Attributes.STATE in update:
+            state = DenonRemote.state_from_avr(update[Attributes.STATE])
+            attributes = helpers.key_update_helper(Attributes.STATE, state, attributes, self.attributes)
+
+        return attributes
+
+    async def _handle_command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        # pylint: disable=R0911
+        match cmd_id:
+            case Commands.ON:
+                return await self._denon_media_player.command(Commands.ON)
+            case Commands.OFF:
+                await self._denon_media_player.command(Commands.OFF)
+            case Commands.TOGGLE:
+                return await self._denon_media_player.command(Commands.TOGGLE)
+
+        if params is None:
+            return StatusCodes.BAD_REQUEST
+
+        if cmd_id == Commands.SEND_CMD:
+            command = params.get("command", "")
+            # TODO: temp fix for toggle command since it's being sent as "remote.toggle" even though it shouldn't
+            if command == "remote." + Commands.TOGGLE:
+                return await self._denon_media_player.command(Commands.TOGGLE)
```

Comment:
Is this wrongfully sent by the core?
I'll check & fix it if that's the case.

Reply:
> nor a native Denon command (contains a dot).

Ooops, not true with the Denon specification below: 0x20 - 0x7F. That includes the dot and lower case characters.
But I could still argue that a `command` should not start with `remote.` and return bad request to immediately fail invalid configurations (or bugs in the core) :-)

Reply:
For me it happens when I use the power button with the remote entity in view. I used the app by albaintor, so maybe there's a bug in the app if you can't reproduce it?

I'll add validation that it shouldn't start with remote. Other than that I'm not sure about stricter validation as the Denon spec says that range is valid. They might not use commands with dots in them right now, but might in the future.

Reply:
Has to be a bug in the app, just ran it on the remote and it works. I'll remove the workaround.